src/lib/loggers.js

```diff
@@ -82,27 +82,10 @@ loggers.error = function() {
     }
 };
 
-/*
- * Robust apply, for IE9 where console.log doesn't support
- * apply like other functions do
- */
 function apply(f, args) {
     if(f && f.apply) {
-        try {
-            // `this` should always be console, since here we're always
-            // applying a method of the console object.
-            f.apply(console, args);
-            return;
-        } catch(e) { /* in case apply failed, fall back on the code below */ }
-    }
-
-    // no apply - just try calling the function on each arg independently
-    for(var i = 0; i < args.length; i++) {
-        try {
-            f(args[i]);
-        } catch(e) {
-            // still fails - last resort simple console.log
-            console.log(args[i]);
-        }
+        // `this` should always be console, since here we're always
+        // applying a method of the console object.
+        f.apply(console, args);
```

Comment:
Could remove this function entirely now, just inline the apply calls. Also the `console.trace` fallback `|| console.log` in log and warn [could be removed](https://developer.mozilla.org/en-US/docs/Web/API/console/trace)

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/5373/commits/bc2b0d112b14db802978c76f3e59aab6e0f87155.